### DISCUSSION TITLE
HASPone v1.03 release

### DISCUSSION THREAD

**Discussion by aderusha** [HASwitchPlate/HASPone#146] [General]
# HASPone v1.03 release

This release is mostly minor bug fixes, feature enhancements, and underlying library updates.

## Changelog

* Add option "Ignore touchevents when backlight is off", allowing one to "secure" a HASPone device if needed by locking out user input and display.  Thanks to @jjfalling for the initial code and idea!
* Add option to set custom discovery prefix at request from user @scyto.  I'll add that this is probably a bad idea and you will likely create a bunch of problems for yourself if you change this value.
* Add configurable value for the number of pages in the Nextion project.  The default value of "11" is what most people will use unless you have modified the Nextion project file.
* Add `platformio.ini` file to simplify library management when compiling HASPone device firmware
* Fix JSON return error with some troubleshooting help from @rebel7580 - thanks!
* Some general code cleanup and small bug fixes
* Update blueprints to avoid logging errors on startup in response to upcoming breaking change in Home Assistant 2021.12
* Update ESP8226 core to 3.0.2
* Update ArduinoJSON library to 6.18.5
* Update MQTT library to 2.5.0
* Update WiFiManager to 2021-10-30 

## Firmware Update Procedure

* Open a browser to your HASPone's IP address or `http://<devicename>.local`
* Click the button labeled `update firmware`
* Click the first button `Update ESP from URL` and allow that process to complete.  The device will power cycle after a minute or two.
* If your HASPone is no longer connected to your computer via a USB cable, uncheck the option `Serial debug output enabled` for a small boost in performance.

## Blueprint Update Procedure

* Navigate to your Home Assistant folder with a file manager
* Replace files under `<homeassistant>/blueprints/automation/HASwitchPlate` with [the blueprints from the repo](https://github.com/HASwitchPlate/HASPone/tree/main/Home_Assistant/blueprints).
* Reload automations or restart Home Assistant

## Links for more HASPone info

* [Main documentation page](https://github.com/HASwitchPlate/HASPone/wiki)
* [HASPone forum](https://github.com/HASwitchPlate/HASPone/discussions)
* [Buy HASPone PCBs](https://www.tindie.com/products/luma/ha-switchplate-hasp-pcb/) - strongly recommended for safe assembly.  [Buy them from me](https://www.tindie.com/products/luma/ha-switchplate-hasp-pcb/) or [download the gerbers](https://github.com/HASwitchPlate/HASPone/blob/main/PCB/HASwitchPlate.gerbers.zip) and send them to your favorite [short-run PCB fab](https://www.allpcb.com/?Mb_InviteId=34099).
* [Buy an assembled device ready to install](https://www.tindie.com/products/luma/ha-switchplate-hasp-single-wide-assembled/) if the build process seems intimidating
* [HASPone Discord chat](https://discord.gg/FvDu8SXSvJ)

**Comment by Petervanoos** [comment]
Thx for the work done. Question. Is it necessary to update all of the blueprints after the firmware upgrade to1.03?

**Reply by aderusha** [reply]
HASPone v 1.03 will work with existing blueprints... but a breaking change in Home Assistant is going to spam errors with the previous version of the blueprints.  Things will still work, but if you'd like to get rid of the errors it'd be best if you upgrade the blueprints before that happens.

**Comment by Petervanoos** [comment]
4 Plates up and running with the new firmware and blueprints. No issues.
